### PR TITLE
Remove occurrences of draft 27/28

### PR DESCRIPTION
**Pull request by JaydenElliott**
# Related Issue #1153

# Changes
* Removed all occurrences of draft 27/28 in both the main and testing code. 
* Removed all code that was only used by drafts 27/28.
* Updated the `invalid_retry()` test to use draft 29.

# Current Problem
The `mitm_retry()` test in neqo-transport/tests/retry.rs is failing at line 415 (the final match statement) because client.state is not switching to closing.

Let me know what I can do to help.

### CODE REVIEW

File: neqo-transport/src/connection/mod.rs [hunk 1]
```diff
@@ -2204,94 +2200,6 @@ impl Connection {
...
-                .map(ConnectionId::as_cid_ref)
-                != tp.map(ConnectionIdRef::from)
-            {
-                return Err(Error::InvalidRetry);
-            }
-        }
-        Ok(())
-    }
-
-    fn validate_cids_draft_28_plus(&mut self) -> Res<()> {
```
Comment (ddragana) [inline neqo-transport/src/connection/mod.rs:2235]: you should not remove this validation. This is still needed for draft 29 to 32 and for final version 1.
You should only remove validate_cids_draft_27.
This is a reason why the test mitm_retry is failing

Comment (martinthomson) [reply neqo-transport/src/connection/mod.rs:2235]: Yep, this is it.  Rename this function as validate_cids(), I think.

**Review by martinthomson — CHANGES_REQUESTED** [review]
Other than the validate_cids() thing, this looks good.  Thanks for putting it together.

File: neqo-transport/src/connection/mod.rs (hunk 1 above)
Comment (JaydenElliott) [reply neqo-transport/src/connection/mod.rs:2235]: Put the function back, renamed it validate_cids() and added the call to the function back to the top of the process_tps(). All tests are now passing on my machine. Squashed my commits, believe its ready!

**Review by ddragana — APPROVED** [review]
looks good.
please resolve conflicts and merge it.

**Comment by JaydenElliott** [comment]
Conflicts resolved. I do not have write access to the main repo so someone else will have to merge.

**Review by martinthomson — APPROVED** [review]

**Comment by martinthomson** [comment]
Thanks @JaydenElliott, I've merged it.

**Pull request merged**